From 19249af11ee65009b40e86838b6f03e5ad0bfdc3 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Mon, 21 Oct 2024 18:50:54 +0200
Subject: [PATCH 918/954] perf(stm32mp2-fdts): set OSPI clock at 266 MHz when
 needed.

Set OSPI clock at 266 MHz on EV1.
Remove the clock on unsupported board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 11260bef53c86e2557baa582835ab44839781d18 perf(stm32mp2-fdts): set OSPI clock at 266 MHz when needed.]
Change-Id: Ia528eb4a54072280584f406e185d3d6683411590
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/479645
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi  | 1 -
 fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi  | 1 -
 fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi | 2 +-
 3 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi b/fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi
index 248a7ae2d0..275fb9f627 100644
--- a/fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi
+++ b/fdts/stm32mp235f-dk-ca35tdcid-rcc.dtsi
@@ -47,7 +47,6 @@
 		FLEXGEN_CFG(4, XBAR_SRC_PLL4, 0, 3)
 		FLEXGEN_CFG(5, XBAR_SRC_PLL4, 0, 2)
 		FLEXGEN_CFG(8, XBAR_SRC_HSI_KER, 0, 0)
-		FLEXGEN_CFG(48, XBAR_SRC_PLL5, 0, 3)
 		FLEXGEN_CFG(51, XBAR_SRC_PLL4, 0, 5)
 		FLEXGEN_CFG(52, XBAR_SRC_PLL4, 0, 5)
 		FLEXGEN_CFG(58, XBAR_SRC_HSE_KER, 0, 1)
diff --git a/fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi b/fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
index 837abb43f3..2b796c8c60 100644
--- a/fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
+++ b/fdts/stm32mp257f-dk-ca35tdcid-rcc.dtsi
@@ -47,7 +47,6 @@
 		FLEXGEN_CFG(4, XBAR_SRC_PLL4, 0, 3)
 		FLEXGEN_CFG(5, XBAR_SRC_PLL4, 0, 2)
 		FLEXGEN_CFG(8, XBAR_SRC_HSI_KER, 0, 0)
-		FLEXGEN_CFG(48, XBAR_SRC_PLL5, 0, 3)
 		FLEXGEN_CFG(51, XBAR_SRC_PLL4, 0, 5)
 		FLEXGEN_CFG(52, XBAR_SRC_PLL4, 0, 5)
 		FLEXGEN_CFG(58, XBAR_SRC_HSE_KER, 0, 1)
diff --git a/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi b/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
index 1f6ded5e01..2ea77e46de 100644
--- a/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
+++ b/fdts/stm32mp257f-ev1-ca35tdcid-rcc.dtsi
@@ -47,7 +47,7 @@
 		FLEXGEN_CFG(4, XBAR_SRC_PLL4, 0, 3)
 		FLEXGEN_CFG(5, XBAR_SRC_PLL4, 0, 2)
 		FLEXGEN_CFG(8, XBAR_SRC_HSI_KER, 0, 0)
-		FLEXGEN_CFG(48, XBAR_SRC_PLL5, 0, 3)
+		FLEXGEN_CFG(48, XBAR_SRC_PLL5, 0, 1)
 		FLEXGEN_CFG(51, XBAR_SRC_PLL4, 0, 5)
 		FLEXGEN_CFG(52, XBAR_SRC_PLL4, 0, 5)
 		FLEXGEN_CFG(58, XBAR_SRC_HSE_KER, 0, 1)
-- 
2.53.0

